src/librustc/hir/lowering.rs

```diff
@@ -2622,9 +2570,18 @@ impl<'a> LoweringContext<'a> {
 
         self.allocate_hir_id_counter(opaque_ty_node_id);
 
+        let input_lifetimes_count = self.in_scope_lifetimes.len() + self.lifetimes_to_define.len();
```

Comment:
This is pretty clever and probably deserves a comment-- that `in_scope_lifetimes` contains all the non-elided lifetime names in scope, and `lifetimes_to_define` says how many lifetimes we have been paired with `CreateParameter`, so the last `lifetime_params.len() - (in_scope_lifetimes + lifetimes_to_define)` elements of `lifetime_params` are all elided lifetimes. Obviously the way I just worded it is probably even more confusing than just reading the code, but I think some explanation would be helpful.